src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -96,8 +109,29 @@ public Task Start(IMessageSession session)
             }
         }
 
+        CancellationTokenSource LongRunningWarning(string message)
+        {
+            var tokenSource = new CancellationTokenSource();
+
+            // ReSharper disable once MethodSupportsCancellation
+            Task.Run(async () =>
+            {
+                try
+                {
+                    await Task.Delay(LongRunningWarningTimeSpan, tokenSource.Token);
+                    Log.Warn(message);
+                }
+                catch (TaskCanceledException)
+                {
+                }
+            });
+
+            return tokenSource;
+        }
+
         IEnumerable<IWantToRunWhenEndpointStartsAndStops> wantToRunWhenBusStartsAndStops;
         ConcurrentBag<IWantToRunWhenEndpointStartsAndStops> thingsRanAtStartup = new ConcurrentBag<IWantToRunWhenEndpointStartsAndStops>();
-        static ILog Log = LogManager.GetLogger<StartableAndStoppableRunner>();
+        public static ILog Log = LogManager.GetLogger<StartableAndStoppableRunner>();
+        public TimeSpan LongRunningWarningTimeSpan { get; set; }
```

Comment:
not a big fan of making properties public for testing reasons. Should we instead define a virtual Log method on the class which can be derived from a testable implementation? Usually a workaround for legacy code, but it would allow us to remove Moq and keep the fields private?